teamup-rocketchat-bot/main.go

```diff
@@ -255,37 +256,121 @@ func checkForMeetings(config *Configuration, chatClient *gorocket.Client) {
 	}
 }
 
+var logOutput string // for initial output
+func init() {
+	// Set the usage for printing usage info
+	flag.Usage = displayHelpMessageWithoutBanner
+	flag.StringVar(&customConfigFile, "config", "", "Point to the configuration (config) file. It overrides the default configuration file located at app directory")
+	flag.StringVar(&customLogPath, "logpath", "", "Set the custom logpath. It overrides the log path specified in configuration (config.yml) file")
+
+	flag.Parse()
+	log.SetPrefix(botPrefix)
+	// If no flags are provided, use defaults
+	fmt.Println(bannerText) // Print the banner, for some  cli awesomeness
+	if len(customConfigFile) == 0 && len(customLogPath) == 0 {
+		customConfigFile = defaultConfigFileName // Set the default config.yml if no parameter was provided for custom config
+		logOutput = "No options specified. Looking for configuration at current directory. Directory for log output will be set at location specified in configuration."
+	} else {
+
+		if len(customConfigFile) == 0 {
+			customConfigFile = defaultConfigFileName
+			logOutput = "Looking for configuration at current directory."
+		} else {
+			logOutput = "Looking for configuration at \"" + customConfigFile + "\"."
+		}
+
+		if len(customLogPath) == 0 {
+			logOutput += " Setting directory specified by configuration for log output."
+		} else {
+			logOutput += " Setting \"" + customLogPath + "\" directory specified by configuration for log output."
+
+		}
+
+	}
+	log.Println(logOutput) // Print for the stdout only
+
+}
+
 func main() {
 
+	// Show proper log message after crash
+	defer func() {
+		if err := recover(); err != nil {
+			log.Printf("The app suffered a panic due to following\n%v", err)
+			log.Println("This is probably related to error mentioned above.")
+			return
+		}
+	}()
+
 	// Setup logger
-	f, err := os.OpenFile(logFile, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644) // 0644 = user can read and write, other and groups can read
+	// Here, we will use default log path and log file name
+	// And only change to provided input log path after validation
+	var defaultPath string
+	// has CustomPath been used
+	var isCustomPath bool = false
+	if len(customLogPath) == 0 {
+		defaultPath = defaultLogFilePath + "/" + defaultLogFileName
+	} else {
+		defaultPath = path.Clean(customLogPath) + "/" + defaultLogFileName
+		isCustomPath = true
+	}
+	defaultLog, err := os.OpenFile(defaultPath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644) // 0644 = user can read and write, other and groups can read
```

Comment:
no need to be so open, `0640` should be enough, same for the file creation further down, or even better have just one `OpenFile` call